erpnext/crm/doctype/appointment_booking_settings/appointment_booking_settings.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+import datetime
+from frappe.model.document import Document
+
+
+class AppointmentBookingSettings(Document):
+	min_date = '01/01/1970 '
+	format_string = "%d/%m/%Y %H:%M:%S"
+
+	def validate(self):
```

Comment:
@0Pranav is this fixed?

Reply:
I'm not sure what the logic for this validation should be.

Reply:
There are two cases...
1. **The number of agents is equal to number of slots**
In this case, the `No. Of Agents` field is not required, as it can be infered from the agents field

2. **The number of agents is not equal to number of slots**
This is also a valid case to be used. Where you don't want all your agents to be booked at the same time, so per slot you can have only 3 bookings. In that case the fieldname can be `No of. bookings per slot`

Depending on what rotation logic you have written (1 or 2) you can make the changess

Reply:
Made changes considering logic 1. Number of Agents field still exists but is hidden. We can move to logic 2 if we need to easier this way.